airbyte-integrations/connectors/source-faker/source_faker/streams.py

```diff
@@ -142,15 +142,15 @@ def read_records(self, **kwargs) -> Iterable[Mapping[str, Any]]:
                 break
 
         self.state = {self.cursor_field: total_records, "seed": self.seed}
-        set_total_user_records(total_records)
 
 
 class Purchases(Stream, IncrementalMixin):
     primary_key = None
-    cursor_field = "user_id"
+    cursor_field = "id"
```

Comment:
Hmm... When I add a primary_key, I loose the choice for an incremntal-deduped sync.  I also was under the impression that the cursor had to be a property which was actually present in the RECRDS (which total_purchase_records wouldn't be).  

Yeah... this is a wierd stream because we generate it like it was a sub-stream, but it can be selected independantly :D